packages/graphback-cli/src/components/generate.ts

```diff
@@ -0,0 +1,271 @@
+import { CliPlugin } from "@test-graphql-cli/common";
+import { CodeFileLoader } from '@graphql-toolkit/code-file-loader';
+import { GraphbackCRUDGeneratorConfig } from "@graphback/core"
+import { migrateDB } from "graphql-migrations"
+import { GeneratedResolvers } from '@graphback/codegen-resolvers/types/api/resolverTypes';
+import Listr, { ListrTask } from 'listr';
+import { prompt } from 'inquirer';
+import chokidar from 'chokidar';
+import debounce from 'debounce';
+import { GraphQLExtensionDeclaration } from 'graphql-config';
+import { print } from 'graphql';
+
+export interface GenerateConfig {
+  folders: {
+    model: string;
+    schema: string;
+    resolvers: string;
+    client: string;
+    migrations: string;
+  };
+  graphqlCRUD: GraphbackCRUDGeneratorConfig;
+  generator: {
+    resolvers: ResolverGeneratorOptions
+    schema: SchemaGeneratorOptions
+    client: { format: 'ts' | 'gql' }
+  }
+  db: { database: string; dbConfig: any; };
+}
+
+const FormatExtensionMap = {
+  'ts': 'ts',
+  'js': 'js',
+  'gql': 'graphql',
+};
+
+export function writeFile(path: string, data: any) {
+  return new Promise<void>(async (resolve, reject) => {
+    await ensureFile(path);
+    fsWriteFile(path, data, err => {
+      if (err) {
+        reject(err);
+      }
+      resolve();
+    })
+  })
+}
+
+export function globPromise(glob: string, options: import('glob').IOptions = {}) {
+  return new Promise<string[]>(async (resolve, reject) => {
+    const { default: globAsync } = await import('glob');
+    globAsync(glob, options, (err, data) => {
+      if (err) {
+        reject(err);
+      }
+      resolve(data);
+    })
+  })
+}
+
+export async function createSchemaFile(cwd: string, generatedSchema: string, config: GenerateConfig) {
```

Comment:
I thought plugins will be responsible for file persistence? It looks like schema, resolver (...etc) generation is still happening in CLI here.

Reply:
I guess this is still working off the old Graphback architecture. Will file-writing logic be removed from here in favour of plugins?

Reply:
Yes.. This is old. I pushed original file for reference to be able to migrate graphql-cli easly and have diff